examples/bri-3/src/bri/identity/bpiSubjects/capabilities/updateBpiSubject/updateBpiSubjectCommand.handler.ts

```diff
@@ -23,17 +22,39 @@ export class UpdateBpiSubjectCommandHandler
         command.id,
       );
 
+    const newPublicKeys = await Promise.all<PublicKey>(
+      command.publicKeys.map(async (key) => {
+        let publicKeyType;
+        switch (key.type.toLowerCase()) {
+          case 'ecdsa':
+            publicKeyType = PublicKeyType.ECDSA;
+            break;
+          case 'eddsa':
+            publicKeyType = PublicKeyType.EDDSA;
+            break;
+          default:
+        }
+
+        const newKey = await this.storageAgent.updatePublicKey(
+          publicKeyType,
+          key.value,
+          bpiSubjectToUpdate.id,
+        );
+
+        return newKey;
+      }),
+    );
+
     this.agent.updateBpiSubject(
```

Comment:
Above code should be hidden behind this method. remmember, code in command handlers explains WHAT happens, not HOW.

Reply:
Fixed.